Repository/PhpcrOdmRepository.php

```
@@ -116,4 +126,87 @@ protected function buildCollection(array $documents)
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentDocument = $this->getManager()->find(null, $resolvedPath);
+
+        if (null === $parentDocument) {
+            throw new InvalidArgumentException(sprintf('Cannot locate parent document at "%s"', $resolvedPath));
+        }
+
+        /** @var PhpcrOdmResource[] $resources */
+        $resources = $resource instanceof IteratorAggregate ? $resource : new ArrayResourceCollection([$resource]);
+        Assert::isInstanceOf($resources, ResourceCollection::class, 'The list should be of instance "ResourceCollection".');
+
+        foreach ($resources as $resource) {
+            Assert::isInstanceOf($resource, CmfResource::class, 'The resource needs to of instance "CmfResource".');
+            Assert::same($resolvedPath, $this->resolvePath($resource->getPath()));
+
+            $this->getManager()->persist($resource->getPayload());
+        }
+
+        $this->getManager()->flush();
+    }
+
+    /**
+     * Moves a resource inside the repository.
+     *
+     * @param string $sourceQuery The Path of the current document.
+     * @param string $targetPath  The parent path of the destination.
+     * @param string $language
+     *
+     * @return int
+     */
+    public function move($sourceQuery, $targetPath, $language = 'glob')
+    {
+        $this->failUnlessGlob($language);
+        Assert::notEq('', trim($sourceQuery, '/'), 'The root directory cannot be moved.');
+
+        $targetPath = $this->resolvePath($targetPath);
+        $sourcePath = $this->resolvePath($sourceQuery);
+
+        $document = $this->getManager()->find(null, $sourcePath);
+        if (null === $document) {
+            throw new \InvalidArgumentException('No document found for source path '.$sourcePath);
+        }
+
+        try {
+            $this->getManager()->move($document, $targetPath);
+            $this->getManager()->flush();
+
+            return 1;
+        } catch (\Exception $e) {
```

I don't think we should swallow exceptions here. Lets remove this try / catch.

Why are we returning 1 / 0?